sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -246,6 +247,68 @@ class Analyzer(
       CleanupAliases)
   )
 
+  /**
+   * For [[UnresolvedAdd]]:
+   * 1. If one side is timestamp/date/string and the other side is interval, turns it to
```

Comment:
it's better to reduce the coupling between analyzer rule and type coercion rule. I think here we should turn into `TimeAdd` if one side is interval, and type coercion rule will cast date/string to timestamp for `TimeAdd`

Reply:
you missed `interval + interval`

Reply:
then how about `if one side is interval and the other side is not interval`?

Reply:
they all go to `ADD` then fail with type checking

Reply:
```
org.apache.spark.sql.AnalysisException
+cannot resolve '(INTERVAL '1 days' + 1)' due to data type mismatch: differing types in '(INTERVAL '1 days' + 1)'
```

Reply:
do we support interval + int?

Reply:
for error reporting, I don't think that there is a perfect solution. We can say that interval should be int to make the operation legal, or int should be timestamp.

Reply:
we do not support cast int to interval, so interval + int is not supported.

Reply:
neither for pg,
```
postgres=# select cast(1 as interval);
ERROR:  cannot cast type integer to interval
LINE 1: select cast(1 as interval);
               ^
postgres=# select cast(1 as date);
ERROR:  cannot cast type integer to date
LINE 1: select cast(1 as date);
               ^
postgres=# select interval '1' day  + 1;
ERROR:  operator does not exist: interval + integer
LINE 1: select interval '1' day  + 1;
                                 ^
```

Reply:
so what's wrong here? pgsql fails and spark fails as well.

Reply:
Er...  here we have no problem, just reply to this comment
>if one side is interval, and type coercion rule will cast date/string to timestamp for TimeAdd

Reply:
type coercion rule will cast date/string to timestamp but not other types, it matches the expected behavior, right?

Reply:
Yes, type coercion will take over after we resolve them

Reply:
Yes, type coercion will take over after we resolve them